File: testerum-backend/testerum-runner/testerum-runner-api/src/main/kotlin/com/testerum/runner/events/model/FeatureStartEvent.kt
package com.testerum.runner.events.model

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDateTime

data class FeatureStartEvent @JsonCreator constructor(
        @JsonProperty("time")         override val time: LocalDateTime = LocalDateTime.now(),
        @JsonProperty("eventKey")     override val eventKey: String,
        @JsonProperty("featureName")  val featureName: String,
        @JsonProperty("tags")         val tags: List<String>
): RunnerEvent

File: old/web/web-frontend/src/app/model/test/event/feature-start.event.ts
import {RunnerEvent} from "./runner.event";
import {RunnerEventTypeEnum} from "./enums/runner-event-type.enum";
import {Serializable} from "../../infrastructure/serializable.model";

export class FeatureStartEvent implements RunnerEvent, Serializable<FeatureStartEvent> {
    time: Date;
    eventKey: string;
    eventType: RunnerEventTypeEnum = RunnerEventTypeEnum.FEATURE_START_EVENT;

    featureName: string;
    tags: Array<string> = [];

    deserialize(input: Object): FeatureStartEvent {
        this.eventKey = input["eventKey"];
        this.time = new Date(input["time"]);
        this.featureName = input["featureName"];
        this.tags = input['tags'] || [];

        return this;
    }

    serialize(): string {
        return undefined;
    }
}
